sips/sip-x/sip-x-nft-metadata.md

```diff
@@ -0,0 +1,341 @@
+
+- `(get-boom-meta () {uri: (string-ascii 35), name: (string-ascii 16), mime-type: (string-ascii 9)})` and
+- `(get-meta? uint {series-id: uint, number: uint, name: (string-utf8 80), uri: (string-ascii 2048), mime-type: (string-ascii 129), hash: (buff 64)})`
+
+### Badges
+
+The [badges contract](https://explorer.stacks.co/txid/0xb874ddbb4a602c22bb5647c7a2f8bfcafbbca7c0c663a175f2270ef3665f33de?chain=mainnet) defines metadata for nfts.
+
+The function signatures for metadata are:
+
+- `(get-badge-meta () {uri: (string-ascii 78111)})` and
+- `(get-meta? (uint) (optional {user: principal}))`
+
+# Activation
+
+This SIP is activated if 5 contracts are deployed that follows this specification. This must happen before Bitcoin tip #750,000.
+
+# Appendix A
+
+List of trait function define in SIPs and specifications specific to these functions
+
+
+| SIP and Trait Function Name | Definition of "existing"| Additional Specification for Properties| Identifier Parameter|
+|-----------------------------|-------------------------|--------------------|-----|
+| SIP-009 nft-trait.get-token-uri | token must be minted and not burnt | NFTs belonging to a group of tokens should use property `properties.collection` of type `string` for the collection name. <br/> Optional property `properties.id` of type `integer` describes the identifier of the token.   | 1st    |
+| SIP-X get-contract-uri      | always | `properties.items` of type array can be used to provide the metadata of all tokens belonging to the collection                  |  X |
+| SIP-010 ft-trait.get-token-uri | always| The required property `decimals` of type `integer` must be the same number as `get-decimals`.               |  X |
+| SIP-013 sip013-semi-fungible-token-trait.get-token-uri | token must be minted and not burnt, no requirements on the number of fungible part of the token|      |  1st |
+
+
+# Appendix B
+
+Attribute types
+
+| Type | Display types | Additional Properties | Comment |
+|------|---------------|-----------------------|---------|
+| Numeric | `number`, `boost_percentage`, `boost_number` | `max_value`| |
+| Date    | `date`     |                       | As unix timestamp in UTC | 
+| String  | empty      |                       |         |
+
+# Appendic C
+
+Common Properties with predefined types.
+
+| Name | Type | Description |
+|------|------|-------------|
+| `collection` | `string` | collection name the token belongs to. See also Appendix A. |
+| `decimals` | `integer` | number of decimals. See also Appendix A. |
+| `id` | `integer` | identifier for NFTs. See also Appendix A. |
+| `created` | `integer` | creation date of the token in unix timestamp | 
+| `symbol`  | `string` | token symbol |
+| `total_supply`| `integer` | number of total supply, e.g. minted tokens |
+| `ip_document_uri` | `string` | link to document about intellectual property (IP) rights |
+| `external_url` | `string` | url that will view the token on an external site |
```

Comment:
another requested field is the `animation_url` which is often for a plain file type, but it can also support a URL to a full html page which is usually embedded in an iframe:

https://docs.opensea.io/docs/metadata-standards

Animation_url also supports HTML pages, allowing you to build rich experiences and interactive NFTs using JavaScript canvas, WebGL, and more. Scripts and relative paths within the HTML page are now supported. However, access to browser extensions is not supported.

Reply:
Has been added